yb-voyager/cmd/importData.go

```diff
@@ -848,39 +848,36 @@ func doOneImport(task *SplitFileImportTask, connPool *tgtdb.ConnectionPool) {
 					}
 				}
 				res, err := conn.PgConn().CopyFrom(context.Background(), reader, copyCommand)
-				rowsCount = res.RowsAffected()
-
-				if err != nil {
+				rowsAffected = res.RowsAffected()
+				
+				if err != nil && strings.Contains(err.Error(), "invalid input syntax") {
+					return false, err
+				}
+				// retry if: err is not nil(except invalid syntax) or the rowsAffected doesn't match
+				if err != nil || (rowsAffected != (task.OffsetEnd - task.OffsetStart)) {
 					log.Warnf("COPY FROM file %q: %s", inProgressFilePath, err)
-					if !strings.Contains(err.Error(), "violates unique constraint") {
+					if err != nil {
 						log.Errorf("RETRYING.. COPY %q FROM file %q due to encountered error: %v ", task.TableName, inProgressFilePath, err)
-						duration := time.Duration(math.Min(MAX_SLEEP_SECOND, math.Pow(2, float64(COPY_MAX_RETRY_COUNT+1-copyRetryCount))))
-						log.Infof("sleep for duration %d before retrying...", duration)
-						time.Sleep(time.Second * duration) // delay for 1 sec before retrying
-						copyRetryCount--
-						return copyRetryCount > 0, err
+					} else {
+						log.Errorf("RETRYING.. COPY %q FROM file %q since rows affected(%d) doesn't match actual row count(%d)",
+							task.TableName, inProgressFilePath, rowsAffected, task.OffsetEnd-task.OffsetStart)
 					}
+					duration := time.Duration(math.Min(MAX_SLEEP_SECOND, 10*float64(COPY_MAX_RETRY_COUNT+1-copyRetryCount)))
+					log.Infof("sleep for duration %d before retrying...", duration)
+					time.Sleep(time.Second * duration)
+					copyRetryCount--
+					return copyRetryCount > 0, err
```

Comment:
Should we rename the `copyRetryCount` as `remainingRetries`?

Reply:
sure, looks more meaningful.